### PR TITLE
chore(stylelint): fix various warnings

### PR DESCRIPTION
**Pull request by SimonFinney**
### Affected issues

- Contributes to #973 
- Contributes to #1084 

### Proposed changes

- Fix remaining Stylelint warnings
- Remove RTL usage in places and unused helpers as this is currently unsupported

### CODE REVIEW

File: src/__tests__/scss/__snapshots__/styles-test.js.snap [hunk 1]
```diff
@@ -23428,7 +23396,9 @@ li.bx--accordion__item--disabled:last-of-type {
   transition: 240ms cubic-bezier(0.2, 0, 0.38, 0.9) background-color;
   margin-left: auto;
   color: var(--cds-inverse-01, #161616);
-  transition: 110ms background-color cubic-bezier(0.2, 0, 0.38, 0.9);
+  transition-duration: 110ms;
```
Comment (thekevinboyle) [inline src/__tests__/scss/__snapshots__/styles-test.js.snap:23399]: Do the props need to be broken out like this now in order to pass the linter?

File: src/components/LayoutModules/DescriptionModule/_index.scss [hunk 2]
```diff
@@ -16,11 +16,14 @@
   &__content {
     @include carbon--type-style($name: body-long-01);
 
-    margin-top: 0;
```
Comment (thekevinboyle) [inline src/components/LayoutModules/DescriptionModule/_index.scss:19]: making a note here to discuss the next time we meet

File: src/components/LayoutModules/DescriptionModule/_index.scss [hunk 3]
```diff
@@ -16,11 +16,14 @@
...
-    margin-bottom: map-get($map: $body-long-01, $key: font-size) *
```
Comment (thekevinboyle) [inline src/components/LayoutModules/DescriptionModule/_index.scss:20]: making a note here to discuss the next time we meet

**Review by thekevinboyle — COMMENTED** [review]
Long time comin, looks good. Made a few notes that we can discuss next time we sync

File: src/__tests__/scss/__snapshots__/styles-test.js.snap (hunk 1 above)
Comment (SimonFinney) [reply src/__tests__/scss/__snapshots__/styles-test.js.snap:23399]: That's right — [transition shorthands must conform as expected](https://github.com/carbon-design-system/stylelint-plugin-carbon-tokens/tree/master/src/rules/layout-token-use#layout-token-use), but individual properties are accepted.

**Pull request merged**